### PR TITLE
Don't make SIR for librustc*

### PR DESCRIPTION
**Pull request by vext01**
Raising to test the time of a full CI run.

No review needed for now.

### CODE REVIEW

**Comment by vext01** [comment]
bors try

File: src/librustc/ty/context.rs [hunk 1]
```diff
@@ -1462,14 +1462,22 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
...
+    /// Get the DefIds of items with MIRs in dependency crates.
+    /// This is for exclusive use by Yorick, as it makes an assumption: that we never want to
+    /// serialise librustc* in the SIR (they are too big and VMs shouldn't use them).
     pub fn metadata_mir_defids(&self) -> DefIdSet {
         let mut res = Vec::new();
         let tbl_lens = self.cstore.def_path_table_lens();
 
         for (cnum, ln) in tbl_lens.iter().enumerate() {
+            let cnum = CrateNum::from_usize(cnum);
+            if self.crate_name(cnum).as_str().starts_with("rustc") {
```
Comment (ltratt) [inline src/librustc/ty/context.rs:1474]: Dumb question: should this be `librustc`?

Comment (vext01) [reply src/librustc/ty/context.rs:1474]: No, I tried that originally and had a `dbg!` inside, which didn't execute. The crate name is without the `lib` prefix.

Comment (ltratt) [reply src/librustc/ty/context.rs:1474]: Did you leave the `dbg!` in and check that this definitely fires as expected?

Comment (vext01) [reply src/librustc/ty/context.rs:1474]: Yeah I did, to check this was having an effect.

From my scrollback:
```
...
Compiling rustc-main v0.0.0 (/home/vext01/research/ykrustc/src/rustc)                                             
[src/librustc/ty/context.rs:1472] "SKIPPING CRATE" = "SKIPPING CRATE"                                                
[src/librustc/ty/context.rs:1472] self.crate_name(cnum) = rustc_binary                                               
[src/librustc/ty/context.rs:1472] "SKIPPING CRATE" = "SKIPPING CRATE"                                                
[src/librustc/ty/context.rs:1472] self.crate_name(cnum) = rustc_std_workspace_core                                   
[src/librustc/ty/context.rs:1472] "SKIPPING CRATE" = "SKIPPING CRATE"                                                
[src/librustc/ty/context.rs:1472] self.crate_name(cnum) = rustc_std_workspace_alloc                                  
[src/librustc/ty/context.rs:1472] "SKIPPING CRATE" = "SKIPPING CRATE"                                                
[src/librustc/ty/context.rs:1472] self.crate_name(cnum) = rustc_demangle                                             
[src/librustc/ty/context.rs:1472] "SKIPPING CRATE" = "SKIPPING CRATE"                                                
[src/librustc/ty/context.rs:1472] self.crate_name(cnum) = rustc_driver                                               [src/librustc/ty/context.rs:1472] "SKIPPING CRATE" = "SKIPPING CRATE"                                                
[src/librustc/ty/context.rs:1472] self.crate_name(cnum) = rustc_save_analysis                                    
...
```

I'm going to think about this over the weekend...

Comment (ltratt) [reply src/librustc/ty/context.rs:1474]: OK.

**Pull request closed without merging**